pull_request_template.md

```diff
@@ -0,0 +1,29 @@
+
+
+# Pre-submission checklist
+
+ * [ ] I read and followed the CONTRIBUTING guidelines.
+ * [ ] I have insured that lint, typecheck, and unit tests complete successfully.
+
+[//]: # (If you can't run the tests locally, create a draft PR to check against the CI pipeline. Once you verify that CI is passing, you can take your PR out of draft status. Please try running the tests locally first, before testing against the CI pipeline.)
+
+
+## Summary of Changes
+
+
+
+
+#### Related Issues, PRs, and Discussions
+
+[//]: # (Please link to related issues, pull requests, and discussions here. If there are no related issues, PRs, or discussions, please provide a justification for the PR.)
+
+
+
+## Docs
+
+* [ ] I have created a pull request to add or update relevant documentation in charmed-hpc/docs or another documentation location.
```

Comment:
A hyperlink might helpful for new contributors who aren't entirely familiar with the project.

```suggestion
* [ ] I have created a pull request to add or update relevant documentation in [charmed-hpc/docs](https://github.com/charmed-hpc/docs) or another documentation location.
```